### ISSUE TITLE
SentryWrappingMiddleware does not expose details of wrapped middleware

### ISSUE THREAD

**Issue by MattFisher** [getsentry/sentry-python#1145]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
Python `sentry-sdk==1.1.0`

### Steps to Reproduce

Use the python sentry sdk alongside the python NewRelic agent (newrelic-python-agent v6.4.4.161).
Wait for slow transaction traces to appear in NewRelic UI.

### Expected Result

The middleware entries in transaction traces should show the names of the middlewares being wrapped.

### Actual Result

All middlewares in transaction traces appear listed as `sentry_sdk.integrations.django_middleware:_wrap_middleware.<locals>.SentryWrappingMiddleware`, as shown in the screenshot below.
<img width="1467" alt="Screen Shot 2021-07-07 at 11 29 21 am" src="https://user-images.githubusercontent.com/70594/124687254-e5592a80-df17-11eb-91cc-b227b108bb95.png">

I've done some investigation, and the NewRelic agent determines the strings to be displayed 
via a `callable_name` function, that bottoms out in a function called [_object_context_py3](https://github.com/newrelic/newrelic-python-agent/blob/v6.4.4.161/newrelic/common/object_names.py#L232). This function returns the module name (via `object.__module__`) and `object.__qualname__` attributes of the passed object.

In the [`sentry_sdk.integrations.django.middleware._wrap_middleware` function](https://github.com/getsentry/sentry-python/blob/1.2.0/sentry_sdk/integrations/django/middleware.py#L177), the `SentryWrappingMiddleware` is given the wrapped middleware's `__name__` attribute.
```python
	if hasattr(middleware, "__name__"):
        SentryWrappingMiddleware.__name__ = middleware.__name__
```
In the standard library's `functools.wraps()`, the wrapping entity is given `('__module__', '__name__', '__qualname__', '__doc__', '__annotations__')` of the wrapped object. This implies it should be pretty safe to do the same with `SentryWrappingMiddleware`, like so:
```python
    if hasattr(middleware, "__name__"):
        SentryWrappingMiddleware.__name__ = middleware.__name__
    if hasattr(middleware, "__module__"):
        SentryWrappingMiddleware.__module__ = middleware.__module__
    if hasattr(middleware, "__qualname__"):
        SentryWrappingMiddleware.__qualname__ = middleware.__qualname__
```
or potentially
```python
    for attr in ('__name__', '__module__', '__qualname__', '__doc__', '__annotations__'):
        if hasattr(middleware, attr):
            setattr(SentryWrappingMiddleware, attr, getattr(middleware, attr))
```

After manually making the first code change listed above, the results were what was expected:
<img width="1197" alt="Screen Shot 2021-07-07 at 11 28 21 am" src="https://user-images.githubusercontent.com/70594/124687233-dbcfc280-df17-11eb-8bb5-7dc2b63a93ad.png">

**Comment by DmytroLitvinov** [comment]
I have faced the same issue :eyes:

**Comment by HazAT** [comment]
@MattFisher thanks for the awesome issue!
Would you mind making a PR out of this since it seems you solved it yourself?
We are happy to merge it and make it better for everyone, wdyt?

**Comment by DmytroLitvinov** [comment]
Hi @HazAT ,
Could you please review related PR https://github.com/getsentry/sentry-python/pull/1202 for that?

**Comment by nicon89** [comment]
We have same problem. Is there any plan for fixing it?

**Comment by sl0thentr0py** [comment]
fixed by #1202, will be in next release
